guppylang-internals/src/guppylang_internals/tracing/object.py

```diff
@@ -539,6 +539,17 @@ def __getitem__(self, item: Any) -> Any:
 
     def to_guppy_object(self) -> GuppyObject:
         state = get_tracing_state()
+        defn = ENGINE.get_checked(self.id)
```

Comment:
this `get_checked` is cheap presumably? jut retrieving something pre-computed?

Reply:
If the function has not been checked before, then this will run type checking, but the same would have happened anyways in the `build_compiled_def` call below